engine/util/src/main/scala/pl/touk/nussknacker/engine/util/loader/ScalaServiceLoader.scala

```diff
@@ -34,12 +32,12 @@ object ScalaServiceLoader {
     }
   }
 
-  def loadNamed[T<:NamedServiceProvider:ClassTag](name: String, classLoader: ClassLoader = Thread.currentThread().getContextClassLoader): T = {
+  def loadNamed[T<:NamedServiceProvider:ClassTag](name: String, default: => List[T] = Nil, classLoader: ClassLoader = Thread.currentThread().getContextClassLoader): T = {
     val available = load[T](classLoader)
     val className = implicitly[ClassTag[T]].runtimeClass.getName
-    Multiplicity(available.filter(_.name == name)) match {
+    Multiplicity(available.filter(_.name == name)).orElse(Multiplicity(default.filter(_.name == name))) match {
```

Comment:
can we just have `available = load[T](classloader) ++ default` ?

Reply:
It would make undistinguishable the two situations:
- when there is a single implementation loaded through SPI and a built-in one, both with the same name;
- when there are multiple implementations loaded through SPI with the same name.
The latter is treated as a misconfiguration.

Reply:
for me it's ok - I mean, I would treat first one also as misconfiguration, I don't see any reason to do that

Reply:
But you may just want to override a  built-in auth method without adding totally new behavior in the frontend app.